tests/test_multipart.py

```diff
@@ -574,6 +589,23 @@ def test_read_chunk_doesnt_breaks_reader(self):
             while not part.at_eof():
                 yield from part.read_chunk(3)
 
+    def test_read_chunk_from_stream_doesnt_breaks_reader(self):
+        reader = aiohttp.multipart.MultipartReader(
+            {CONTENT_TYPE: 'multipart/related;boundary=":"'},
+            Stream(b'--:\r\n'
+                   b'\r\n'
+                   b'test'
+                   b'\r\n--:\r\n'
+                   b'\r\n'
+                   b'passed'
+                   b'\r\n--:--'))
+        while True:
+            part = yield from reader.next()
+            if part is None:
+                break
+            while not part.at_eof():
+                yield from part.read_chunk(4)
```

Comment:
How many chunks do you expect to read here? Assuming that the behavior of `test_read_chunk_from_stream_doesnt_breaks_reader` test is the same as for `test_read_chunk_by_length_doesnt_breaks_reader` it should be 3: `b'test'`, `b'pass'`, `b'ed'`. However, here you produce `b'test'`, `b''`, `b'pass'`, `b'ed'`. 

Since multipart boundary must be started from the new line, you eventually ate that delimiter and added it to the body it doesn't have.